ark-testing/src/daemon/mod.rs

```diff
@@ -96,58 +91,51 @@ where T : DaemonHelper + Send + Sync + 'static
 
 	pub async fn try_start(&mut self) -> anyhow::Result<()> {
 		self.inner.make_reservations().await?;
+
 		let mut cmd = self
 			.inner
 			.get_command()
 			.await?;
 
 		cmd
 			.stdout(Stdio::piped())
-			.stderr(Stdio::piped())
-			.kill_on_drop(true);
-
-		// Start listening to logs for init-messages
-		let mut stdout_rx = self.stdout_sender.lock().await.subscribe();
-		let mut stderr_rx = self.stderr_sender.lock().await.subscribe();
-
-		let init_future = self.inner.wait_for_init(&mut stdout_rx, &mut stderr_rx);
+			.stderr(Stdio::piped());
 
 		trace!("{}: Trying to spawn {:?}", self.inner.name(), cmd);
 		let mut child = cmd.spawn()?;
 
-		// process stdout
-		let stdout_sender = self.stdout_sender.clone();
-		let mut stdout_reader = BufReader::new(child.stdout.take().expect("Child has stdout")).lines();
-
-		tokio::task::spawn(async move {
-			loop {
-				match stdout_reader.next_line().await.context("Failed to read line")? {
-					None => return Ok::<(), anyhow::Error>(()),
-					Some(line) => {
-						stdout_sender.lock().await.send(line.to_string())?;
-					}
+		let stdout = child.stdout.take().unwrap();
+		let stderr = child.stderr.take().unwrap();
+		let stdout_log = self.stdout_handler.clone();
+		let stderr_log = self.stderr_handler.clone();
+
+		self.stdout_jh = Some(std::thread::spawn(move || {
+			let reader = BufReader::new(stdout);
+			for line in reader.lines() {
+				let line = line.unwrap();
+				for handler in stdout_log.lock().unwrap().iter_mut() {
+					handler.process_log(&line)
 				}
 			}
-		});
-
-		// Read stdrr
-		let stderr_sender = self.stderr_sender.clone();
-		let mut stderr_reader = BufReader::new(child.stderr.take().expect("Child has stderr")).lines();
-
-		tokio::task::spawn(async move {
-			loop {
-				match stderr_reader.next_line().await? {
-					None => return Ok::<(), anyhow::Error>(()), //Reached the end
-					Some(line) => {
-						stderr_sender.lock().await.send(line.to_string())?;
-					}
+		}));
+
+		self.stderr_jh = Some(std::thread::spawn(move || {
+			let reader = BufReader::new(stderr);
+			for line in reader.lines() {
+				let line = line.unwrap();
+				trace!("Processing line stderr: {}", line);
+				for handler in stderr_log.lock().unwrap().iter_mut() {
+					trace!("Handled");
+					handler.process_log(&line)
```

Comment:
Is this stuff modeled after the original deamon-runner impl? Btw, that trace log will be fun :D

Reply:
Oops, I shouldn't have comitted those logs.

Reply:
I have removed the lines